src/node.cc

```diff
@@ -847,12 +860,20 @@ void PromiseRejectCallback(PromiseRejectMessage message) {
 
   if (event == v8::kPromiseRejectWithNoHandler) {
     callback = env->promise_reject_unhandled_function();
+
+    // Exit the process in case the unhandled promise is garbage collected
+    // without being handled later on.
+    Persistent<Promise>* p = new Persistent<Promise>(env->isolate(), promise);
+    p->SetWeak(p, WeakCallback, WeakCallbackType::kFinalizer);
+
     value = message.GetValue();
 
     if (value.IsEmpty())
       value = Undefined(isolate);
   } else if (event == v8::kPromiseHandlerAddedAfterReject) {
     callback = env->promise_reject_handled_function();
+    Persistent<Promise>* p = new Persistent<Promise>(env->isolate(), promise);
+    p->ClearWeak();
```

Comment:
Hm, I think my original comment was slightly misleading – we don’t really need to create a new `Persistent` handle and call `ClearWeak()` on it. But we do need to do *something* here, to clean up the original persistent handle that we created. (I think mentioning `ClearWeak()` was just a mistake on my side, sorry about that.)

In particular, we need to `delete` that handle, in this case, otherwise each `new Persistent` expression will allocate memory without it ever being free’d.